### ISSUE TITLE
Test migration behaviour against client with fresh FF install

### ISSUE THREAD

**Issue by tublitzed** [mozilla-services/services-engineering#46]
After running through [these tests](https://github.com/mozilla-services/services-engineering/issues/45), we want to check against the following scenario:

When a user with [this collection of bookmarks gets migrated](https://github.com/mozilla-services/services-engineering/blob/master/scripts/bookmarks/samples/mocks/med.json), what happens when they login for the very first time on a brand new sync client?

This can go one of two ways:

a) A sync is performed, data is fetched from spanner and populated on device.
b) A sync is performed, empty dataset that exists in fresh install _replaces_ existing dataset, data is lost.

**Comment by erkolson** [comment]
Looks like it hasn't synced with the migrate py node yet:
```
MySQL [tokenserver]> select * from users where email = 'e9389413141b45709b90c773e7674218@api.accounts.firefox.com';
+-----------+---------+-----------------------------------------------------------+---------------+----------------------------------+---------------+---------------+--------+-----------------+
| uid       | service | email                                                     | generation    | client_state                     | created_at    | replaced_at   | nodeid | keys_changed_at |
+-----------+---------+-----------------------------------------------------------+---------------+----------------------------------+---------------+---------------+--------+-----------------+
| 129848941 |       1 | e9389413141b45709b90c773e7674218@api.accounts.firefox.com | 1586203503210 | ae5bb8a9e99a997a86352a9fcad377d3 | 1572293538351 | 1586203504034 |    550 |            NULL |
| 142092891 |       1 | e9389413141b45709b90c773e7674218@api.accounts.firefox.com | 1586203503211 | 073730744458fbe00e256caa564d202c | 1586203504034 | 1586205140841 |    550 |   1586203503211 |
| 142094861 |       1 | e9389413141b45709b90c773e7674218@api.accounts.firefox.com | 1586203503211 | 073730744458fbe00e256caa564d202c | 1586205140841 |          NULL |    836 |            NULL |
+-----------+---------+-----------------------------------------------------------+---------------+----------------------------------+---------------+---------------+--------+-----------------+
```

**Comment by tublitzed** [comment]
^ For completeness sake, Erik and I talked offline and client is now synced as of this AM.

**Comment by tublitzed** [comment]
Unfortunately, I found the following:

1. I first uploaded a collection from my test account, success sync log is [here](https://drive.google.com/open?id=1miZkb_st-cobWSrTcnNQgDTRBg5H1xln)
2. @erkolson migrated the user.
3. I signed in from a fresh install on a new device (iOS). I synced, and got no data.
4. Then, I signed in from a fresh profile on Desktop. I synced and also got no data. [log here](https://drive.google.com/file/d/17iP66f_z_M4f-rcXpFge8xoaVRpPUxuX/view?usp=sharing)
5. THEN, I signed in from my existing profile on Desktop (that I'd used to originally sync all my data), tried to sync and got errors. [Sync error log here](https://drive.google.com/open?id=1MeHfSX7N39XMeyMDq4hTafB0Ih5xcJhB)

cc @pjenvey

**Comment by pjenvey** [comment]
After [initial issues w/ migrating individual users](https://github.com/mozilla-services/syncstorage-rs/issues/568#issuecomment-613137600) were resolved, I was seeing a fresh desktop install syncing for the first time not seeing all its bookmarks.

This was due to a bug in `migrate_node.py` where not all data was successfully migrated to Spanner. Fixed in #586
